Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h

```diff
@@ -87,6 +88,46 @@ class LinkDecorationFilteringController {
     WeakHashSet<LinkDecorationFilteringDataObserver> m_observers;
 };
 
+class StorageAccessPromptQuirkObserver : public RefCounted<StorageAccessPromptQuirkObserver>, public CanMakeWeakPtr<StorageAccessPromptQuirkObserver> {
+public:
+    ~StorageAccessPromptQuirkObserver() = default;
+
+private:
+    friend class StorageAccessPromptQuirkController;
+
+    explicit StorageAccessPromptQuirkObserver(Function<void()>&& callback)
+        : m_callback { WTFMove(callback) }
+    {
+    }
+
+    static Ref<StorageAccessPromptQuirkObserver> create(Function<void()>&& callback)
```

Comment:
I think this class would look better with this public and the invokeCallback() too, as mentioned in my earlier review. Without a "friend".